R/class2tree.R

```diff
@@ -31,6 +31,17 @@
 #' @details See [vegan::taxa2dist()]. Thanks to Jari Oksanen for
 #' making the taxa2dist function and pointing it out, and Clarke & Warwick
 #' (1998, 2001), which taxa2dist was based on.
+#' The taxonomy tree created is not only based on the clustering of the taxonomy
+#' ranks (e.g. strain, species, genus, ...), but it also utilizes the actual 
+#' taxon clades (e.g. mammals, plants or reptiles, etc.). The process of this 
+#' function is as following: First, all possible taxonomy ranks and their 
+#' corresponding IDs for each given taxon will be collected from the input.
+#' Then, the rank vectors of all taxa will be aligned, so that they together 
+#' will become a matrix with columns are a list of ordered taxonomy ranks of all
+#' given taxa and rows are the rank vectors of those taxa. After that, the rank
+#' matrix will be converted into ID matrix, any missing rank will have a pseudo
+#' ID from the previous rank. Finally, this ID matrix will be used to cluster
+#' taxa that have similar taxonomy string together.
```

Comment:
Unclear here. What is a "taxonomy string"?

Reply:
Like:
Taxon A: strain 11, species 21, genus 31, family 41,..., superkingdom xy,..
Taxon B: species 22, genus 32, family 41, ..., superkingdom xy,...
I think I will replace it by "taxonomy hierarchy". Do you think it will be more understandable?

Reply:
yeah, string doesn't include what it contains, whereas hierarchy does